internal_ws/ykcompile/src/lib.rs

```diff
@@ -595,7 +599,8 @@ impl TraceCompiler {
     /// Copy bytes from one memory location to another.
     fn copy_memory(&mut self, dest: &RegAndOffset, src: &RegAndOffset, size: u64) {
         // We use memmove(3), as it's not clear if MIR (and therefore SIR) could cause copies
-        // involving overlapping buffers.
+        // involving overlapping buffers. XXX this sounds like a big thing not to be sure about!
+        // what's the underlying problem?
```

Comment:
Using `memcpy` would require that the equivalent of something like `foo[0..2] = foo[1..3]` in MIR is UB.

Reply:
And it's not UB, right?

Reply:
It hasn't been decided yet as far as I know, but I would personally expect it to become UB. At least in most cases. Keeping `memmove` for now in case it doesn't become UB is fine though.

Reply:
If we expect it to become UB I'd be fine with sticking an `assert` in there to catch such cases and moving away from `memmove`. I don't feel strongly about this, though.

Reply:
It can only be realistically asserted at runtime when executing the compiled trace.

Reply:
Yes, that's the only thing that's making me not be gun-ho about it. On reflection, I think we should probably just link to a Rust github issue / RFC for the relevant UB (if we know one!) in a comment.

Reply:
https://github.com/rust-lang/rust/issues/68364 is the issue about this.

Reply:
Perfect, thanks!

Reply:
Fixed in db5a334.